### ISSUE TITLE
(fit vs fit_generator) and (sequential vs functional) same results?

### ISSUE THREAD

**Issue by sidrat28** [keras-team/keras#10401]
I am trying to train a deep neural network using transfer learning in Keras with tensorflow. There are different ways to do that, if your data is small you can afford computing features using the pre-trained model for the entire data and then use those features to train and test a small network, this is good as you don't need to compute those features for each batch and at each epoch. However, if the data is large, it will be impossible to compute features for the entire data, in this case we use `ImageDataGenerator`, `flow_from_directory` and `fit_generator`. In this case features are computed each time fore each batch at each epoch which make things much slower. I was assuming that both approaches produce similar results in terms of accuracy and loss. The problem is that I took a small data-set and tried both approaches and got completely different results. I will appreciate if someone can tell if something is wrong in the provided code and/or why I am getting different results please?

Approach when having large data-set: 

    from keras.applications.inception_v3 import InceptionV3,preprocess_input
    from keras.preprocessing.image import ImageDataGenerator
    from keras.models import Model

    datagen= ImageDataGenerator(preprocessing_function=preprocess_input)
    
    train_generator = datagen.flow_from_directory('data/train', 
                                                  class_mode='categorical',
                                                  batch_size=64,...)
    vaild_generator = datagen.flow_from_directory('data/valid',
                                                  class_mode='categorical',
                                                  batch_size=64,...)
    
    base_model = InceptionV3(weights='imagenet', include_top=False)
    x = base_model.output
    x = Conv2D(filters = 128 , kernel_size = (2,2)) (x)
    x = MaxPooling2D()(x)
    x = GlobalAveragePooling2D()(x)
    x = Dense(1024, activation='relu')(x)
    predictions = Dense(2, activation='softmax')(x)
    model = Model(inputs=base_model.input, outputs=predictions)
    
    for layer in base_model.layers:
        layer.trainable = False
    
    model.compile(optimizer='rmsprop', loss='categorical_crossentropy',...)
    
    model.fit_generator(generator = train_generator,
                        steps_per_epoch = len (train_generator),
                        validation_data = valid_generator ,
                        validation_steps = len(valid_generator),
                        ...)`
    
Approach when having small data-set:

    from keras.applications.inception_v3 import InceptionV3,preprocess_input
    from keras.models import Sequential
    from keras.utils import np_utils

    base_model = InceptionV3(weights='imagenet', include_top=False)
    train_features = base_model.predict(preprocess_input(train_data))
    valid_features = base_model.predict(preprocess_input(valid_data))
    
    model = Sequential()
    model.add(Conv2D(filters = 128 , kernel_size = (2,2),
                     input_shape=(train_features [1],
                                  train_features [2],
                                  train_features [3])))
    model.add(MaxPooling2D())
    model.add(GlobalAveragePooling2D())
    model.add(Dense(1024, activation='relu'))
    model.add(Dense(2, activation='softmax'))     
    
    model.compile(optimizer='rmsprop', loss='categorical_crossentropy',...)
    
    model.fit(train_features, np_utils.to_categorical(y_train,2), 
              validation_data=(valid_features, np_utils.to_categorical(y_valid,2)),
              batch_size=64,...)

**Comment by sgladstein** [comment]
I am having a similar issue. Did you ever solve this problem?
